sahi/utils/coco.py

```diff
@@ -1406,6 +1414,47 @@ def get_area_filtered_coco(self, min=0, max=float("inf"), intervals_per_category
 
         return area_filtered_coco
 
+    def get_coco_with_proper_bboxes(self):
+        """
+        Limits overflowing bounding boxes to image dimensions.
+        """
+
+        coco = Coco(
+            name=self.name,
+            image_dir=self.image_dir,
+            remapping_dict=self.remapping_dict,
+            ignore_negative_samples=self.ignore_negative_samples,
+        )
+        coco.add_categories_from_coco_category_list(self.json_categories)
+
+        if self.images is not None:
+            image_id_to_height = {}
+            image_id_to_width = {}
+            for coco_img in self.images:
+                if coco_img.annotations is not None:
```

Comment:
we should also add the images that does not contain any annotations

Reply:
@ssahinnkadir bu ife hicbir zaman girmiyor cunku annotations degiskeninin default degeri `[]`, kaldirabiliriz bu satiri o yuzden.